docs/onnxruntime_dependencies.dot

```diff
@@ -0,0 +1,88 @@
+digraph "GG" {
+  compound=true;
+	
+node [
+  fontsize = "12"
+];
+subgraph cluster_0 {
+    label = "onnxruntime.dll";
+    "ort_graph" [ label="onnxruntime_graph\n(schemas)" shape="box"];
+    "ort_common" [ label="onnxruntime_common" shape="box"];
+    "ort_util" [ label="onnxruntime_util" shape="box"];
+    "ort_mlas" [ label="onnxruntime_mlas" shape="box"];
+    "ort_optimizer" [ label="onnxruntime_optimizer" shape="box"];    
+    "ort_session" [ label="onnxruntime_session" shape="box"];
+    "ort_graph" -> "ort_common"
+    "onnx" [ label="onnx" shape="box"];
+    "protobuf" [ label="Google Protobuf" shape="box"];
+    "onnx" -> "protobuf"
+    "ort_graph" -> "protobuf"    
+    "ort_graph" -> "onnx"
+    "ort_optimizer" -> "onnx"        
+    "ort_framework" [ label="onnxruntime_framework" shape="box"];
+    "ort_framework" -> "ort_graph" 
+    "ort_framework" -> "ort_common" 
+    "ort_framework" -> "onnx" 
+    "ort_cpu_provider" [ label="onnxruntime_cpu_provider\n(kernels)" shape="box"];
```

Comment:
other than cpu and cuda, should any other providers be listed? (i.e. DML or others from intel/nvidia/etc)

Reply:
Yes I can. But the graph will be more complex. 
How many we should list there?
Please give me the names.

Reply:
I think it's quite readable in its current state.
We don't need to list every possible dependency/optional components.
EP's is a pretty easy concept to understand, and they are more about extensions to the core. so we can leave other ones out of this (especially externally developed ones)